src/Cli/dotnet/commands/dotnet-sln/remove/Program.cs

```diff
@@ -77,7 +76,18 @@ private async Task RemoveProjectsAsync(string solutionFileFullPath, IEnumerable<
             foreach (var projectPath in projectPaths)
             {
                 var project = solution.FindProject(projectPath);
-                if (project != null)
+                // If the project is not found, try to find it by name without extension
+                if (project is null && !Path.HasExtension(projectPath))
+                {
+                    try 
+                    {
+                        project = solution.SolutionProjects.SingleOrDefault(p => Path.GetFileNameWithoutExtension(p.DisplayName) == projectPath);
+                    }
+                    catch {
+                        project = null;
```

Comment:
Shouldn't need anything in this catch. Also, what kind of exception is going to be caught here? It's often best to avoid having the exception in the first place, but if you need it, it's good to be specific.

In this case, I'm assuming you're just looking for an exception from SingleOrDefault? Except that, when provided with a default, should just return the default, so there shouldn't be an exception from the Single part. So can you just get rid of the try/catch?

Reply:
It throws an exception if more than one matches are found

Reply:
How about replacing this with .Where and looking at how many elements are returned?